spec/controllers/invitations_controller_spec.rb

```diff
@@ -194,39 +261,191 @@
+        invitation.invitee_email = old_email
+        subject
+
+        it_redirects_to_with_error(new_user_session_path, "Sorry, you don't have permission to access the page you were trying to reach. Please log in.")
+        expect(invitation.invitee_email).to eq(old_email)
+      end
     end
   end
 
   describe "DELETE #destroy" do
-    it "does not allow non-admins to destroy" do
-      invitation = create(:invitation)
-      fake_login
-      delete :destroy, params: { id: invitation.id }
+    let(:invitation) { create(:invitation) }
+    subject { delete :destroy, params: { id: invitation.id } }
 
-      it_redirects_to_with_error(
-        user_path(controller.current_user),
-        "Sorry, you don't have permission to access the page you were trying to reach."
-      )
+    authorized_roles = InvitationPolicy::EXTRA_INFO_ROLES + %w[legal]
+
+    authorized_roles.each do |admin_role|
+      context "when logged in as an admin with role #{admin_role}" do
+        before do
+          admin.update!(roles: [admin_role])
+          fake_login_admin(admin)
+        end
+
+        context "when invitation creator is a user" do
+          it "deletes invitation and redirects to user invitations path" do
+            invitation_creator = create(:user)
+            invitation.creator = invitation_creator
+            subject
+
+            it_redirects_to_with_notice(user_invitations_path(invitation_creator), "Invitation successfully destroyed")
+          end
+        end
+
+        context "when invitation creator is an admin" do
+          it "deletes invitation and redirects to admin invitations path" do
+            invitation_creator = create(:admin)
+            invitation.creator = invitation_creator
+            subject
+
+            it_redirects_to_with_notice(admin_invitations_path(), "Invitation successfully destroyed")
+          end
+        end
+
+        it "errors if invitation fails to destroy" do
+          allow_any_instance_of(Invitation).to receive(:destroy).and_return(false)
+          subject
+
+          expect(flash[:error]).to match("Invitation was not destroyed.")
+        end
+      end
     end
 
-    it "allows admin to delete invitations" do
-      invitation = create(:invitation)
-      admin.update!(roles: ["policy_and_abuse"])
-      fake_login_admin(admin)
- 
-      delete :destroy, params: { id: invitation.id }
-      it_redirects_to_with_notice(admin_invitations_path, "Invitation successfully destroyed")
+    context "when logged in as an admin with no role" do
+      before do
+        admin.update!(roles: [])
+        fake_login_admin(admin)
+      end
+
+      it "redirects with error and does not delete the invitation" do
+        admin.update!(roles: [])
+        fake_login_admin(admin)
+        subject
+
+        it_redirects_to_with_error(new_user_session_path, "Sorry, you don't have permission to access the page you were trying to reach. Please log in.")
+        expect(invitation).to exist
```

Comment:
This check tries to call `exist?` or `exists?` on the object which I don't think does anything sensible (it's not defined for the record in Rails). Try making a `find` for the invitation instead, then the exist check will hit [a Rails method](https://guides.rubyonrails.org/active_record_querying.html#existence-of-objects).